### DISCUSSION TITLE
Middleware to redirect to login route then back to the original route?

### DISCUSSION THREAD

**Discussion by nbigaouette** [tokio-rs/axum#2758] [Q&A]
### Summary

Is it possible to have:
1. a middleware that checks the request for specific information (f.e. presence of a cookie) and based on that either allow the request through or redirect to a "login" route
2. the login route redirecting to whatever route redirected it there, if any, with the original request data?

What I'm trying to do is verify that every requests contains a valid cookie and create it if absent. I want to uniquely identify clients: I said "login" above but there is no auth, it's just a unique id that the server sets for each clients.

Thanks!

### axum version

0.7.5

**Comment by mladedav** [comment]
You can return a 3xx status code from your middleware without running the handlers themselves and you can craft the location so that you encode all the data needed into the URL.

This won't work for bodies though as those might be just too large. A lot of this might depend on your concrete use-case.
